Error: `for` must be an `id` reference to a form control and may not reference other element types. To create the intended relationship, the `div` must have a `role` of "group" and an `aria-labelledby` attribute that references the `label`. Or use `fieldset` and `legend`.
Error: Use `aria-describedby` to reference a group description.
Error: For usability, it is recommended to put the group description at the top of the component, just after the label.
Error: Checkboxes in a group should use a `name` value that ends with "[]" to allow server-side code to collect the values as an array (or similar).
Error: `aria-invalid="true"` is used to accessibly indicate fields that have errors.
Error: Color alone should not be used to convey errors.
Error: The error text should not be hidden when the component has an error. Note: If the checkbox control is not nested in its `label`, wrapping the two elements within another element (such as a `div`) is preferable. Having a shared parent element gives designers more control over the eventual layout of the checkbox group than a `br` does.
diff --git a/output/Checkbox Group with Validation Error/4/51fed43c-5824-4060-bd9e-e545b382a6e7.html b/output/Checkbox Group with Validation Error/4/51fed43c-5824-4060-bd9e-e545b382a6e7.html
index 4fc9f22..64e4a41 100644
--- a/output/Checkbox Group with Validation Error/4/51fed43c-5824-4060-bd9e-e545b382a6e7.html	
+++ b/output/Checkbox Group with Validation Error/4/51fed43c-5824-4060-bd9e-e545b382a6e7.html	
@@ -1,10 +1,16 @@
-<label for="fruits">What fruits do you like?</label>
-<div id="fruits">
-  <input type="checkbox" id="bananas" name="fruits" value="bananas">
-  <label for="bananas">Bananas</label>
-  <input type="checkbox" id="oranges" name="fruits" value="oranges">
-  <label for="oranges">Oranges</label>
-  <input type="checkbox" id="apples" name="fruits" value="apples">
-  <label for="apples">Apples</label>
-</div>
-<p class="error" style="color: red; display: none;">You must choose one or more fruits</p>
\ No newline at end of file
+<label id="fruits-label">What fruits do you like?</label>
+<div role="group" aria-labelledby="fruits-label" aria-describedby="fruits-error">
+  <p class="error" id="fruits-error">You must choose one or more fruits</p>
+  <div>
+    <input type="checkbox" id="bananas" name="fruits[]" value="bananas" aria-invalid="true">
+    <label for="bananas">Bananas</label>
+  </div>
+  <div>
+    <input type="checkbox" id="oranges" name="fruits[]" value="oranges" aria-invalid="true">
+    <label for="oranges">Oranges</label>
+  </div>
+  <div>
+    <input type="checkbox" id="apples" name="fruits[]" value="apples" aria-invalid="true">
+    <label for="apples">Apples</label>
+  </div>
+</div>
\ No newline at end of file
